Classes/Notifications/NotificationClient.swift

```diff
@@ -123,4 +127,49 @@ final class NotificationClient {
         })
     }
 
+    func fetchWatchedRepositories(completion: @escaping (Result<[Repository]>) -> Void) {
+        guard let viewer = githubClient.userSession?.username else {
+            completion(.error(nil))
+            return
+        }
+
+        githubClient.request(GithubClient.Request(
+            path: "users/\(viewer)/subscriptions"
+        ) { (response, _) in
+            // https://developer.github.com/v3/activity/watching/#list-repositories-being-watched
+            if let jsonArr = response.value as? [ [String: Any] ] {
+                var repos = [Repository]()
+                for json in jsonArr {
+                    if let repo = Repository(json: json) {
+                        repos.append(repo)
+                    }
+                }
+                completion(.success(repos.sorted { $0.name < $1.name }))
+            } else {
+                completion(.error(response.error))
+            }
+        })
+    }
+
+    // MARK: Private API
+
+    func markNotifications(repo: (owner: String, name: String)?, completion: @escaping (Bool) -> Void) {
```

Comment:
Looks pretty similar to https://github.com/rnystrom/GitHawk/pull/1291/files#diff-d08adcb8b7b2ae5fc2e0f5abfaa92a42R34, maybe you can share some logic?